code/modules/paperwork/photocopier.dm

```diff
@@ -1,223 +1,301 @@
+/obj/item/stock_parts/circuitboard/photocopier
+	name = "circuitboard (photocopier)"
+	build_path = /obj/machinery/photocopier
+	board_type = "machine"
+	origin_tech = "{'engineering':1, 'programming':1}"
+	req_components = list(
+			/obj/item/stock_parts/printer/buildable = 1,
+			/obj/item/stock_parts/manipulator       = 2,
+			/obj/item/stock_parts/scanning_module   = 1,
+		)
+	additional_spawn_components = list(
+		/obj/item/stock_parts/console_screen      = 1,
+		/obj/item/stock_parts/keyboard            = 1,
+		/obj/item/stock_parts/power/apc/buildable = 1
+	)
+
+////////////////////////////////////////////////////////////////////////////////////////
+// Photocopier
+////////////////////////////////////////////////////////////////////////////////////////
 /obj/machinery/photocopier
-	name = "photocopier"
-	icon = 'icons/obj/bureaucracy.dmi'
-	icon_state = "photocopier"
-	var/insert_anim = "photocopier_animation"
-	anchored = 1
-	density = 1
-	idle_power_usage = 30
-	active_power_usage = 200
-	power_channel = EQUIP
-	atom_flags = ATOM_FLAG_NO_TEMP_CHANGE | ATOM_FLAG_CLIMBABLE
-	obj_flags = OBJ_FLAG_ANCHORABLE
-	var/obj/item/copyitem = null	//what's in the copier!
-	var/copies = 1	//how many copies to print!
-	var/toner = 30 //how much toner is left! woooooo~
-	var/maxcopies = 10	//how many copies can be copied at once- idea shamelessly stolen from bs12's copier!
+	name                  = "photocopier"
+	icon                  = 'icons/obj/machines/photocopier.dmi'
+	icon_state            = "photocopier"
+	anchored              = TRUE
+	density               = TRUE
+	idle_power_usage      = 30
+	active_power_usage    = 200
+	atom_flags            = ATOM_FLAG_NO_TEMP_CHANGE | ATOM_FLAG_CLIMBABLE
+	obj_flags             = OBJ_FLAG_ANCHORABLE
+	construct_state       = /decl/machine_construction/default/panel_closed
+	maximum_component_parts = list(
+		/obj/item/stock_parts/printer = 1,
+		/obj/item/stock_parts         = 10,
+	)
+	uncreated_component_parts = null
+	var/tmp/insert_anim   = "photocopier_animation"
+	var/obj/item/scanner_item                  //what's in the scanner
+	var/obj/item/stock_parts/printer/printer   //What handles the printing queue
+	var/tmp/max_copies    = 10                 //how many copies can be copied at once- idea shamelessly stolen from bs12's copier!
+	var/tmp/busy          = FALSE              //Whether we should allow people to mess with the settings and contents
+	var/accept_refill     = FALSE              //Whether we should handle attackby paper to be sent to the paper bin, or to the scanner slot
+	var/total_printing    = 0                  //The total number of pages we are printing in the current run
 
-/obj/machinery/photocopier/interface_interact(mob/user)
-	interact(user)
+/obj/machinery/photocopier/Initialize(ml)
+	. = ..()
+	if(.!= INITIALIZE_HINT_QDEL && ml && printer)
+		//Mapped photocopiers shall spawn with ink and paper
```

Comment:
If so, the correct check is the `populate_parts` arg, not mapload.

Reply:
Gotcha